### ISSUE TITLE
dplug:canvas rare crash

### ISSUE THREAD

**Issue by p0nce** [AuburnSounds/Dplug#471]
Some shapes causes a crash within the rasterizer.

Probably the assert in https://github.com/AuburnSounds/Dplug/blob/master/canvas/dplug/canvas/rasterizer.d#L371

When it does, it's reproducible however I failed to extract the shape at the time.

**Comment by p0nce** [comment]
Found with random triangle a minimal example that crash the rasterizer:
```d
import dplug.canvas;
import dplug.graphics;
import dplug.graphics.drawex;

void main() 
{
    OwnedImage!RGBA image = new OwnedImage!RGBA(512, 512);
    ImageRef!RGBA imageRef = image.toRef();

    Canvas canvas;
    canvas.initialize(imageRef);

    {
        float x1 = -4.48518;
        float y1 = 1.69193;
        float x2 = 9.89451;
        float y2 = 0.150122;
        float x3 = -1.97198;
        float y3 = 1.42315;

        canvas.beginPath();
            canvas.moveTo(x1, y1);
            canvas.lineTo(x2, y2);
            canvas.lineTo(x3, y3);
            canvas.fill();      
    }
}
```

**Comment by p0nce** [comment]
Further discussed in https://github.com/cerjones/dg2d/issues/2

**Comment by p0nce** [comment]
Work-arounded in v9.2.3

**Comment by p0nce** [comment]
Fix ported from upstream in v9.2.4
